packages/documentsearch-extension/src/index.ts

```diff
@@ -100,6 +104,28 @@ const extension: JupyterFrontEndPlugin<ISearchProviderRegistry> = {
     const nextCommand: string = 'documentsearch:highlightNext';
     const prevCommand: string = 'documentsearch:highlightPrevious';
 
+    if (settingRegistry) {
+      const loadSettings = settingRegistry.load(extension.id);
+      const updateSettings = (settings: ISettingRegistry.ISettings): void => {
+        const searchDebounceTimeSetting = settings.get('searchDebounceTime')
+          .composite as number;
+        if (searchDebounceTime) {
+          searchDebounceTime = searchDebounceTimeSetting;
```

Comment:
It will not be taken into account before the user close and reopen the search box. Should we document it in the settings description or close search box opened?

Reply:
That condition does not make sense. I have removed it.